20150228-wordpress-guids.md

```diff
@@ -0,0 +1,57 @@
+# Post GUIDs: Sometimes You Should Update Them
+
+***TL;DR** &mdash; Updating the GUIDs of a WordPress site that's already live will annoy people subscribed to its RSS feeds. But if the site has never been live, you should update the GUIDs.*
+
+![Snowflake](https://s3.amazonaws.com/cdn.deliciousbrains.com/content/uploads/2015/03/14002042262_f43b4f2e9b_o-1024x576.jpg)
+
+*[Vega, snowflake desktop wallpaper](https://www.flickr.com/photos/chaoticmind75/14002042262/) by Alexey Kljatov. [CC BY-NC 2.0](https://creativecommons.org/licenses/by-nc/2.0/)*
+
+#### What is a GUID?
+
+GUID is an acronym for Globally Unique Identifier. It is a concept that goes well beyond WordPress and is used in all kinds of applications. In the past, I used GUIDs heavily working with ASP.NET and Microsoft SQL Server for example. Those GUIDs looked like this:
+
+```
+6F9619FF-8B86-D011-B42D-00C04FC964FF
+```
+
+(I know, shocking right! A Microsoft stack, really dude? If it's any consolation, I hated every minute of it.)
+
+RSS feeds are another example that employ GUIDs. Every item in an RSS feed can have a `<guid>` element.
+
+#### How does WordPress use GUIDs?
+
+WordPress employs GUIDs when it generates its RSS (and Atom) feeds. If you take a look at the [RSS feed of this blog](https://deliciousbrains.com/blog/feed/) you will see elements similar to this:
+
+```
+<guid isPermaLink="false">https://deliciousbrains.com/?p=6806</guid>
+```
+
+WordPress uses a real, working post URL as the GUID. In fact, if you go to the URL in the example above, it will redirect you to the blog post it belongs to. This is also the URL the post has when using the default Permalink setting. Since URLs are globally unique by their nature, they make pretty good GUIDs so long as they don't change. And in this case, since we're using the post ID in the URL, it's almost certain not to change. A good design decision.
```

Comment:
There's a possible opportunity to reference the post about not merging databases here, based on the fact that a merge might change the ID of a post when the ID has been used in both databases.